util/systemd/systemd-integration.sh.in

```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+#
+# Note this script deliberately requires bash as it uses several bash-isms
+
+TIMEOUT_USEC=$(cat /run/systemd/reboot-to-boot-loader-menu)
+let "TIMEOUT = ($TIMEOUT_USEC + 500000) / 1000000"
```

Comment:
I believe upstream GRUB requires all the shipped scripts to be valid POSIX shell and not contain any bash-ism. I think is OK to merge this for Fedora but it might be another thing to cleanup before posting the patches.

Another option is to use arithmetic expansions, i.e:

```
TIMEOUT=$(((TIMEOUT_USEC + 500000) / 1000000))
```
I know this is much uglier but since you had upstreaming in mind maybe is worth to change it.

Reply:
Hi Javier, I agree with your suggestion. I've just completed implenting and testing it; and pushed out a new version with the suggested changes.

So I believe that this should be ready for merging now.